inference-engine/src/vpu/common/src/ngraph/transformations/merge_subsequent_dsr_operations.cpp

```diff
@@ -22,6 +22,7 @@ MergeSubsequentDSROperations::MergeSubsequentDSROperations() {
         }
         // this will create a new DSR with correct inputs
         auto newDsr = dsr->copy_with_new_inputs({predecessor->input_value(0), dsr->input_value(1)});
+        newDsr->set_friendly_name(dsr->get_friendly_name());
```

Comment:
I don't think it's a good idea. You see, the motivation for creating `newDSR` is to able to use *simple* `for`-loop below. I think it's better not introduce new operation (and so don't handle its name) and just add appropriate condition for body loop (replace all consumers except ourselves)

Reply:
As I wasn't involved to the discussion of this problem, I'm not fully in the context, but as I understand, we can just replace the producer of `dsr` first (so that `dsr` is no more consumer of `predecessor`), and do the same loop without any conditions.
I checked the model which it fixed and it works fine with this approach.
@GlebKazantaev @KorA6 please take a closer look at this.

Reply:
[Link ](https://github.com/openvinotoolkit/openvino/pull/4022#discussion_r576171041)to the discussion.